CMakeLists.txt

```diff
@@ -33,9 +33,17 @@ link_libraries(${ROOT_LIBRARIES} ${ROOT_COMPONENT_LIBRARIES})
 add_definitions(${ROOT_DEFINITIONS})
 add_definitions("-DMONITORING")
 
+# Set up C++ Standard
+# ``-DCMAKE_CXX_STANDARD=<standard>`` when invoking CMake
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "")
+
+if(NOT CMAKE_CXX_STANDARD MATCHES "17|20")
+  message(FATAL_ERROR "Unsupported C++ standard: ${CMAKE_CXX_STANDARD}")
+endif()
```

Comment:
Hi Thomas, thanks again. The comment here is the same as elsewhere. The default standard is fine, but the restriction should be avoided to allow for overrides at build time.

Reply:
Hi @andresailer , yes, in this case, most particularly cmake configuration. You can set the standard with an additional flag, however, this conditional expression would then veto the use of, say, C++23, so I think this if block should be avoided.

Reply:
I mainly put this validation here to avoid some potential issues. We can adjust that to whatever you deem suitable (or remove it entirely). The main point with this series of pull requests was to switch to `CMAKE_CXX_STANDARD` for setting the c++ standard as that is the recommended way to do it (and is also what many build tools use to set it) instead of putting it into the `CMAKE_CXX_FLAGS`. Would you accept that simplified set of changes?

Reply:
Hi @tmadlener, yes, the updates to use CMAKE_CXX_STANDARD with a default of 17 are completely fine, it's really only the few lines highlighted in the comment that are problematic (plus the requested style change in the SDK PR), so if those are removed the remaining changes should be fine.